File: apps/apple/TopsyTurvy/TopsyTurvyToolchainTests/TokenResult.swift
/// Represents the result of a `topsyturvy_tokens` request from the Topsy Turvy toolchain.
///
/// Mirrors `TokenResult` in `operetta/BWHazel.TopsyTurvy.Embedded/Analysis/TokenResult.cs`. Property names
/// match the JSON contract's PascalCase keys exactly, since `EmbeddedJsonContext` uses no naming policy.
struct TokenResult: Decodable {
    /// The recognised token spans, in source order.
    let Tokens: [TokenInfo]
}

File: operetta/BWHazel.TopsyTurvy.Embedded/Analysis/TokenResult.cs
using System.Collections.Generic;

namespace BWHazel.TopsyTurvy.Embedded.Analysis;

/// <summary>
/// Represents the result of a tokenise request, returned as JSON by <see cref="NativeExports.ToolchainExports.GetTokens"/>.
/// </summary>
/// <param name="Tokens">The recognised token spans, in source order.</param>
public record TokenResult(IReadOnlyList<TokenInfo> Tokens);
